content/blog/sivayanama-simplifying-open-api-specification-with-reusable-common-data-models.md

```diff
@@ -0,0 +1,114 @@
+---
+title: Simplifying OpenAPI specification with reusable common data models
+date: 2023-08-26T08:06:19.141Z
+author: BalaSubramanian Vetrivel
+authorimage: /img/vetrivel-balasubramanian1-photo.jpg
+disable: false
+tags:
+  - API, OpenAPI, OpenAPI specification, OpsRamp API, Yaml
+---
+The [OpenAPI specification](https://www.openapis.org)  is one of the most widely followed API contracts. It is language-agonistic.
+With the help of these OpenAPI specifications, clients can understand the API and invoke them without having access to the code or worrying about the implementation details. 
+
+At times, this open specification file is becoming too complex to manage and understand. In this article, I will discuss the techniques to simplify the open specification with loosely coupled reusable data models.
+
+## Items of OpenAPI specification
+OpenAPI specifications files have many definitions in them. However, the below list of entities are typically bigger in terms of the number of lines and tends to be reused in the specification file.
+
+- schemas
+- pathitems
+- parameters
+- requestBodies
+- responses
+- headers
+- examples
+- linkscallbacks
+
+I will discuss four approaches to deal with these definitions.
+
+## In line definition
+The definition is inline right at the reference point as shown below.
```

Comment:
Consider modifying to:
"With inline definition, the definition is inline right at the reference point, as shown below:"